Fix homekit_controller on climate-1.0

Jc2k opened the pull request:
## Description:

This fixes the tests for homekit_controller climate on the climate-1.0 branch. It also fixes a typo in climate (humitidy vs humidity)

**Related issue (if applicable):**  #23899

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

Jc2k committed:
Fix tests on climate-1.0 branch

Jc2k committed:
As part of climate-1.0, make state return the heating-cooling.current characteristic

pvizeli commented on homeassistant/components/homekit_controller/climate.py:
```diff
@@ -166,14 +164,6 @@ def _update_relative_humidity_target(self, value):
-
-        if self._state == HVAC_MODE_OFF and \
-                self._current_mode != HVAC_MODE_OFF:
-            return HVAC_MODE_OFF
```
Remove the state overwrite completely because that is something that the base entity object do

Jc2k replied:
Done

pvizeli reviewed (changes requested):
```
   @property
    def hvac_mode(self):
        """Return current operation ie. heat, cool, idle."""
        return self._current_mode
```

That is wrong. If you have a current action they match with `CURRENT_*` use `hvac_action` property.

Jc2k committed:
Fixes from review

Jc2k commented:
Re: `hvac_mode` - the docstring in homekit_controller was wrong, and the variable names were confusing but the implementation was doing the right thing i think so i have:

 * Fixed the docstrings based on what is in climate now
 * Renamed `_current_mode` to `_target_mode` and documented what Apple think it means at the point where it is populated
 * `_current_mode` is now actually set to the current operating mode (it was previously getting put in self._state) with the `CURRENT_` stuff
 * Have added support for `hvac_action`
 * Added a test to show `hvac_mode` and `hvac_action` are independent and not getting conflated in homekit_controller.

The homekit tests are still passing with these changes, and flake8 and pylint is passing for the homekit bits too.
